### DISCUSSION TITLE
[Feature Request] Request signature with parameters

### DISCUSSION THREAD

**Discussion by 128keaton** [docusealco/docuseal#48] [Ideas]
Essentially, I'd like to create a link to send to a user with the form pre-filled and just waiting for their signature. Some sort of API request to generate an unsigned but filled document ready to send to the user would be handy. Something like Webhooks (which has already been suggested) would make this app super powerful and much closer to a drop-in replacement for Docusign

Related
https://github.com/docusealco/docuseal/discussions/25
https://github.com/docusealco/docuseal/discussions/40
https://github.com/docusealco/docuseal/discussions/39

**Comment by 128keaton** [comment]
For example, how this process works in Docusign is essentially you have a Clickwrap template (which is basically a PDF waiting to be signed with lots of fields, options, etc) that can be pre-filled and sent to the user. 

When you need a signature from a user on the document, you call the Docusign API endpoint with the Clickwrap ID, the parameters to fill out the form, the email address to which the document signature request will be sent to, and lastly the webhooks to fire upon signing/not signing.

**Comment by kios19** [comment]
did you get any help with this?

**Comment by AlexBTurchyn** [comment]
@128keaton @kios19 In the last release (1.1.3), I added the ability to set default field values via API. Let me know if the new feature is useful for you.

**Reply by kios19** [reply]
```plaintext
would it be possible to share some type of documentation for setting it, or point me in the right direction, im trying to initiate one , and im posting  the payload as: 
{  
       "template_id": 1,
       "emails":  "dg@gmail.com",
        "values": [
       {
         "field": "date",
         "value": "2023-08-09"
       },
       {
         "field": "operating_name",
         "value": "Test company"
       },
       {
         "field": "id",
         "value": "333333"
       }
     ]
   }
as raw data but it doesn't seem to fill up those values
```

**Reply by kios19** [reply]
never mind, had not seen the new update, thank you for that
